### ISSUE TITLE
UnboundLocalError: local variable 'new_value' referenced before assignment

### ISSUE THREAD

**Issue by Asutu** [NBISweden/EMBLmyGFF3#34]
Hi,

in testing `EMBLmyGFF3` to prepare an embl file with genome annotation for submission to EMBL,  I'm getting the following error:

```
EMBLmyGFF3 scaffold1.gff scaffold1.fa --topology linear --transl_table 1 --molecule_type 'genomic DNA' --species 'Salix viminalis' --locus_tag TEST --project_id PRJEB00001 --de 'Single-molecule assembly' -o scaffold1.embl

    #############################################################################
    # NBIS 2018 - Sweden                                                        #
    # Authors: Martin Norling, Niclas Jareborg, Jacques Dainat                  #
    # Please visit https://github.com/NBISweden/EMBLmyGFF3 for more information #
    #############################################################################

12:25:01 WARNING feature: Unknown qualifier 'makerName' - skipped              ]
12:25:01 WARNING feature: Unknown qualifier '_QI' - skipped
12:25:01 WARNING feature: Unknown qualifier '_AED' - skipped
12:25:01 WARNING feature: Unknown qualifier '_eAED' - skipped
12:25:01 ERROR qualifier: local variable 'new_value' referenced before assignment
Traceback (most recent call last):
  File "/opt/pyenv/versions/2.7.10/envs/EMBLmyGFF3_venv/lib/python2.7/site-packages/EMBLmyGFF3/modules/qualifier.py", line 88, in _by_value_format
    formatted_value=new_value
UnboundLocalError: local variable 'new_value' referenced before assignment
```

I'm just using one scaffold for testing purposes. The `head` of the GFF file looks like this:

```
##gff-version 3
scaffold1	repeatmasker	match	2	581	896	+	.	ID=scaffold1:hit:709:1.3.0.0;Name=species:rnd-4_family-62|genus:Unspecified;Target=species:rnd-4_family-62|genus:Unspecified 253 760 +
scaffold1	maker	gene	444	3240	.	+	.	ID=salix_viminalisG00000000001;Name=at3g47200_37;makerName=genemark-scaffold1-processed-gene-0.6
scaffold1	maker	mRNA	444	3240	.	+	.	ID=salix_viminalisT00000000001;Parent=salix_viminalisG00000000001;Dbxref=PFAM:PF03140,InterPro:IPR004158;Name=at3g47200_37;_AED=0.25;_QI=0|0|0|0.25|1|1|4|0|412;_eAED=0.25;makerName=genemark-scaffold1-processed-gene-0.6-mRNA-1;product=UPF0481 protein At3g47200
scaffold1	maker	exon	444	618	.	+	.	ID=salix_viminalisE00000000001;Parent=salix_viminalisT00000000001;makerName=genemark-scaffold1-processed-gene-0.6-mRNA-1:1
scaffold1	maker	CDS	444	618	.	+	0	ID=salix_viminalisC00000000001;Parent=salix_viminalisT00000000001;makerName=genemark-scaffold1-processed-gene-0.6-mRNA-1:cds
scaffold1	maker	exon	827	912	.	+	.	ID=salix_viminalisE00000000002;Parent=salix_viminalisT00000000001;makerName=genemark-scaffold1-processed-gene-0.6-mRNA-1:2
scaffold1	maker	CDS	827	912	.	+	2	ID=salix_viminalisC00000000001;Parent=salix_viminalisT00000000001;makerName=genemark-scaffold1-processed-gene-0.6-mRNA-1:cds
scaffold1	maker	exon	2167	3072	.	+	.	ID=salix_viminalisE00000000003;Parent=salix_viminalisT00000000001;makerName=genemark-scaffold1-processed-gene-0.6-mRNA-1:3
scaffold1	maker	CDS	2167	3072	.	+	0	ID=salix_viminalisC00000000001;Parent=salix_viminalisT00000000001;makerName=genemark-scaffold1-processed-gene-0.6-mRNA-1:cds
```

Note that I have a local `translation_gff_feature_to_embl_feature.json` mapping "match" to "repeat_region":

```
 "match": {
 	"target": "repeat_region"
 },
```

The head of the fasta file is as:

```
>scaffold1
TAAAATAAAAAAAATCGGGTCGGGCCCAACTAAATGGGCCGGCTAGCCAAGATGGGCCAAAGCCCAATTTTAATGGGCTGGGCCGAGAGCCGTCCAGCCC
AAGACACGCAGAAGAGGAAAAAAAGAAAAGGGCAAAACGGCACTGTTTAGCACATGTTAATTAAACAGTTTACGTATTTCGTGAACAGTAAAATGGTGGT
CGGCCGACCACGACGAGAGGGTCACCTGCTATTGCCGCCGGGTAGAGGAGGTCGAGGTGGTTGTCCCTGTGGTTGTGGAGTCGAAAATGGTGGCCCGTGG
CGGCCGGAGGAGGCGTTGGAAGTGGCCGGTCTGTTGCTTGCTGTCGTCGGGGCTGTCACTGTTTCTTCGCCGGAGAGGACGACTGAGCTGCTGGAGTTGA
GGGGAGGCTGAAGGTGGTGATGAGGGTGGATATGGGTGGTTGAATGGTGGCTGTTGGAGGAGAGAGAGAGACGCCGGGTCCTCTGGTTTTAGAGAGAGAA
TGCTGTCGGGGAGAGAGAAAGGAGCTGCAACAGGCTGAGAGACGAAGGAGAGAGAGAGAGAAAGGGCTGCTGTGTCGCCGGAGCTGGAGAGGAAGAAAGG
GTGGCTGCCTCTGCGTGTGTATGCTTGTGTTCTGCAAATTTACCACGTCTTCGTCTTCCTCCTCCAGCCTTAATTTGAAACTGAAACTAAAATATTCGCC
TCTGTTCTCTCAAAACTTCTCAGTTTCTTCCTTGCTTTTCTTTGCCCAAATTTCTGTCGATTTTCCTCCCGTTTTTTCTCCCTTCTTTCTCCCCCTTCTG
CATGCATTTCATGCATGTATTTATAGGTTTGAAAGGCAACCCTTCAGCTGCCCATGGCGTGCAGCGAAGGGTTGCCGCCTGTGATTGCAGGTGGCGTGCC
```

Also, I installed `EMBLmyGFF3 `  in a virtual environment, but when I run the maker example with the command `EMBLmyGFF3-maker-example`, I do get the correct output `EMBLmyGFF3-maker-example.embl` and without errors.

I can't seem to spot possible formatting errors in the input files but I'm just using this tool for the first time so there may be something that I'm missing.

Any help would be kindly appreciated.
Many thanks,
Pedro

**Comment by Juke34** [comment]
Could you relaunch with '-vvv' option to have more verbose information ?
Thx

**Comment by Asutu** [comment]
Of course, please attached log file.
[EMBLmyGFF3.test.log](https://github.com/NBISweden/EMBLmyGFF3/files/2952225/EMBLmyGFF3.test.log)

**Comment by Juke34** [comment]
Ok then it's related to a bug we are aware of: #32
We shouldn't stop the tool but just raise a WARNING.
It's because the KEGG database (from the dbxref attribute) is not among accepted database at ENA.
Here the list of databases accepted: [https://www.ncbi.nlm.nih.gov/genbank/collab/db_xref/](https://www.ncbi.nlm.nih.gov/genbank/collab/db_xref/)

A way would be to remove those information or edit the legal_dbxref.json to add those DB among the list of accepted one. In the second case you will create an EMBL file that do not strictly follow the ENA specification and I'm no sure how ENA will handle it.

Another thing, I realised recently that ENA do not keep DB_xref information when it's processed by their pipeline. I'm still waiting for information about it from them, but the only information I got is: "We get rid of DB_xref information and redo our own functional information"... 
At the end I don't really know how the db_xref data is handle by ENA.

**Comment by Asutu** [comment]
Great, so I guess the safest way for now is to remove this attribute from the GFF. Will try to do that.
Many thanks for the prompt support.

**Comment by Juke34** [comment]
As your annotation comes from NBIS I guess several other DB reported in the dbxref attribute are not supported by ENA. Here is the list of DB we can report that are not accepted by ENA:  
 "KOG",
 "KEGG",
 "MetaCyc",
 "Reactome",
 "UniPathway"

**Comment by Asutu** [comment]
Removing these values from the dbxref solves this error.

**Comment by Juke34** [comment]
fixed in release 1.2.5
